### PR TITLE
Add cargo deny init argument.

### PR DESCRIPTION
**Pull request by foresterre**
Makes it possible to initialize a cargo deny configuration file.
~~The initialization command currently supports setting a few options directly:~~
* ~~unlicensed~~
* ~~copyleft~~
* ~~allow-osi-fsf-free~~
* ~~confidence-threshold~~

Of course, additional options can still be set by editing the cargo deny config
with an editor.

It is also possible to create a config file other than the default
<context>/deny.toml (idea copied from the cargo deny check arguments) by providing `--config <path-to-file>`.

### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

I worked on the init subcommand, which can be used to create a basic cargo-deny configuration file.

### Related Issues

related: #42 (Add "init" subcommand) 

### Draft

Before this draft PR is finished:

~~I intend to at least add support for the allow and deny fields as options e.g.
`cargo deny init --allow MIT Apache-2.0 ISC --deny  AGPL-3.0-only AGPL-3.0-or-later`~~

I also intend to add a test suite; however since none exists (for subcommands) this requires a few careful considerations and will take me a bit more time :). 

Happy to make changes or add other additions if they are desired, in this PR or additional PRs :smiley:

### CODE REVIEW

File: src/cargo-deny/init.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    // make sure the path does not terminate in '..'; we need a file name.
+    ensure!(
+        &cfg_file.file_name().is_some(),
+        "unable to create a config file with the given name ; the given file path is not valid"
+    );
+
+    let config: InitConfig = args.into();
+    let content = toml::to_string(&config).context("unable to create config (toml) file")?;
+
+    std::fs::write(cfg_file, content).context("unable to write config file")?;
```
Comment (Jake-Shadle) [inline src/cargo-deny/init.rs]: Would be nice to log the path of the file that was created here.

**Comment by foresterre** [comment]
Thanks for the feedback! 🙂 

I've removed the arguments (except for the config file path) and replaced it with a commented configuration file. The comments are based on (or copied where applicable) the comments from the readme. What do you think? 🙂

**Comment by Jake-Shadle** [comment]
Nice! I guess I would just say to remove the comments on the actual example values so that they stand out more from the explanatory comments, then you don't need the additional guide comment at the top.

**Comment by foresterre** [comment]
> Nice! I guess I would just say to remove the comments on the actual example values so that they stand out more from the explanatory comments, then you don't need the additional guide comment at the top.

I am a bit confused which comments the "comments on the actual example values" are 🙄.
Do you mean the examples below `[ban]` (1) and the `[[licenses.clarify]]` example (2)? Or do you mean all comments (currently) starting with `--`?

(1) https://github.com/EmbarkStudios/cargo-deny/pull/48/commits/57bb79d40f2171f2275afd9bc8c26812efef8f06#diff-c012aa2026af44810754ffb835299323R120
(2) https://github.com/EmbarkStudios/cargo-deny/pull/48/commits/57bb79d40f2171f2275afd9bc8c26812efef8f06#diff-c012aa2026af44810754ffb835299323R75

**Comment by Jake-Shadle** [comment]
Yes, I mean remove all the comments except the ones with `--` then the user has a somewhat real file, that they will be able edit immediately to add remove things based on their needs, will the additional explanatory comments as inline help.

**Comment by foresterre** [comment]
I cleaned up the template. Now each comment just gives an overview of what a config option means. Pre-filled values for lists have been removed. This was inspired by the template of a fresh `.zshrc` configuration file.

I see the following advantages:
* easier to review the configuration by having less noise
* having a single place where details need to be changed if they'll ever change (namely still the readme)

I still wasn't entirely sure what you mean with your previous comment; hopefully I understood correctly that you wanted the template config to contain less noise.
If you want something changed; I am happy to make the changes. Thanks for the fast feedback :).

In addition: I think it would be better to add the tests to a separate PR, which can also work towards fixing #1.

**Comment by Jake-Shadle** [comment]
Yah, this is much more concise but still gives enough info that a new user can get a feel for what is available. And you have a link back to the README for more in depth info, so I'm pretty happy with this!

**Comment by Jake-Shadle** [comment]
Ahh sorry, apparently github doesn't give a notification when a PR goes out of WIP. :stuck_out_tongue:
